Package.swift

```diff
@@ -1348,3 +1305,170 @@ func googleAppMeasurementDependency() -> Package.Dependency {
 
   return .package(url: appMeasurementURL, exact: "10.12.0")
 }
+
+func abseilDependency() -> Package.Dependency {
+  let packageInfo: (url: String, range: Range<Version>)
+
+  // If building Firestore from source, abseil will need to be built as source
+  // as the headers in the binary version of abseil are unusable.
+  if ProcessInfo.processInfo.environment["FIREBASE_SOURCE_FIRESTORE"] != nil {
+    packageInfo = (
+      "https://github.com/firebase/abseil-cpp-SwiftPM.git",
+      "0.20220623.0" ..< "0.20220624.0"
+    )
```

Comment:
The source based abseil versions are equivalent to that of binary abseil's version range. 

These binary abseil has a major version of 1 and adds two zeros to the minor. 
_Lower bound: 0.20220623.0 → >1<.20220623>00<.0 → 1.2022062300.0_
_Upper bound: 0.20220624.0 → >1<.20220624>00<.0 → 1.2022062400.0_